fastcrypto-vdf/src/math/modular_sqrt.rs

```diff
@@ -35,27 +35,29 @@ pub fn modular_square_root(
         return Err(InvalidInput);
     }
 
+    let two = BigInt::from(2);
+
     let a = a.mod_floor(p);
     match mod8(p) {
         3 | 7 => Ok(a.modpow(&((p + 1) >> 2), p)),
         5 => {
             let mut x = a.modpow(&((p + 3) >> 3), p);
-            let c = x.modpow(&2.into(), p);
+            let c = x.modpow(&two, p);
             if c != a {
-                x = x * BigInt::from(2).modpow(&((p - 1) >> 2), p) % p;
+                x *= two.modpow(&((p - 1) >> 2), p) % p;
             }
             Ok(x)
         }
         1 => {
-            let mut d: BigInt = 2.into();
+            let mut d: BigInt = two.clone();
             while jacobi::jacobi(&d, p).expect("p is positive and odd") != -1 {
                 d += 1;
                 if &d >= p {
                     return Err(InvalidInput);
                 }
             }
             let p_minus_1: BigInt = p - 1;
-            let s = p_minus_1.trailing_zeros().expect("p > 1");
+            let s = p_minus_1.trailing_zeros().expect("p is not 1");
```

Comment:
what if zero?

Reply:
p != 0 is checked earlier (line 15). I'll expand the error message to make it clear.